### ISSUE TITLE
Number typeguards

### ISSUE THREAD

**Issue by jods4** [microsoft/TypeScript#41569]
**TypeScript Version:**  4.1.1-rc

**Search Terms:** Number typeguard

**Code**

```ts
let x = 42 as number | null;
Number.isFinite(x) && x.toString()
```

**Expected behavior:** No error nor warnings.
The following built-in Number functions should be defined as typeguards as they can only return true when their argument is a number:
`isFinite`, `isInteger`, `isNaN`, `isSafeInteger`.
They currently typed as simple `boolean` and hence do not modify TS flow analysis.

**Actual behavior:** TS2531, objet is possibly null

**Playground Link:** https://www.typescriptlang.org/play?ts=4.1.0-beta#code/DYUwLgBAHhC8EBYBMECGBnCA7ArgWwCMQAnCAH2x2GAG4AoAOXyOIDoBLdAMXa3bBAAKKAEoIAMnHRWYAPYBlMMV4BzQSJpA

**Related Issues:**

**Comment by MartinJohns** [comment]
Duplicate of #39090. Search terms used: `isFinite`

**Comment by jods4** [comment]
Indeed.
